### PR TITLE
Support deprecated message and fields

### PR DESCRIPTION
**Pull request by abn**
Initial attempt at warning on deprecated field use.

Generated sample:
```py
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# sources: deprecated.proto
# plugin: python-betterproto
import warnings
from dataclasses import dataclass

import betterproto


@dataclass
class Test(betterproto.Message):
    """Some documentation about the Test message."""

    v: int = betterproto.int32_field(1)
    value: int = betterproto.int32_field(2)

    def __post_init__(self) -> None:
        warnings.warn("Test is deprecated", DeprecationWarning)
        super().__post_init__()
        if self.v:
            warnings.warn("Test.v is deprecated", DeprecationWarning)
```

### CODE REVIEW

**Comment by boukeversteegh** [comment]
Great work! Unfortunately the OOP refactor has made this PR in conflict.
Would you be willing to port it to the new structure?

**Comment by abn** [comment]
@boukeversteegh this should be ready again.

File: tests/inputs/deprecated/deprecated.proto [hunk 1]
```diff
@@ -0,0 +1,9 @@
+syntax = "proto3";
+
+// Some documentation about the Test message.
+message Test {
+    // Some documentation about the value.
+    option deprecated = true;
+    int32 v = 1 [deprecated=true];
```
Comment (boukeversteegh) [inline tests/inputs/deprecated/deprecated.proto:7]: Is `int32 v = 1 [deprecated=true]` redundant with the `deprecated_field` test?

Comment (abn) [reply tests/inputs/deprecated/deprecated.proto:7]: It was done specificaly to verify that we handle the case where the message and the field deprecation explicitly.

**Comment by adriangb** [comment]
Happy to see that this worked well with that refactor 😄

**Pull request merged**